code/game/objects/items/rcd/RPD.dm

```diff
@@ -257,6 +278,10 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 
 /obj/item/pipe_dispenser/dropped(mob/user, silent)
 	UnregisterSignal(user, COMSIG_MOUSE_SCROLL_ON)
+	if(!(upgrade_flags & RPD_UPGRADE_SPEEDPIPE))
```

Comment:
you should invert this (don't early return) so that `..()` is still called at the end